src/recode.jl

```diff
@@ -168,13 +168,8 @@ function recode!(dest::CategoricalArray{T}, src::AbstractArray, default::Any, pa
     dest
 end
 
-function recode!(dest::CategoricalArray{T, N, R}, src::CategoricalArray,
-                 default::Any, pairs::Pair...) where {T, N, R<:Integer}
-    if length(dest) != length(src)
-        throw(DimensionMismatch("dest and src must be of the same length " *
-                                "(got $(length(dest)) and $(length(src)))"))
-    end
-
+function _recode!(dest::CategoricalArray{T, N, R}, src::CategoricalArray,
+                 default::Any, pairs::Tuple) where {T, N, R<:Integer}
     vals = T[p.second for p in pairs]
     if default === nothing
         srclevels = levels(src)
```

Comment:
Below there's a `firsts` variable which should be renamed to `recode_from` and moved to the top of the function for consistency.